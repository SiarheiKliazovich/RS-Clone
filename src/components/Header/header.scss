@import "../../sass/mixins";
@import "../../sass/variables";

.header {
  position: fixed;
  width: 100%;
  top: 0;
  background-color: $color-accent;
  box-shadow: $box-shadow;
  transition: $transition;
  z-index: 10;
}

.header__wrapper {
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.header__hamburger {
  @include icon-background('../../assets/images/svg/hamburger.svg', 32px);
  min-width: 32px;
  min-height: 32px;
  cursor: pointer;
}

.header__logo {
  transition: 0.5s;

  &:hover {
    transform: scale(1.1);
  }

  @include icon-background('../../assets/images/png/logo.png', 50px, 200px);
  margin: 15px;
  cursor: pointer;

  @include media-small-mobile {
    width: 150px;
    height: 40px;
  }
}

.header__list {
  display: flex;
  justify-content: end;
  align-items: center;
  gap: 30px;
  flex-grow: 2;

  @include media-small-mobile {
    gap: 20px;
  }
}

.header__profile {
  @include icon-background('../../assets/images/jpg/user.jpg', 50px);
  border-radius: $border-radius-sm;
  cursor: pointer;
  z-index: 80;
}

.language {
  display: flex;
  align-items: flex-start;
  gap: 10px;
  color: $font-color-white;
  cursor: pointer;
}

.language__icon {
  display: inline-block;
  width: 100%;
  height: 100%;
}

.language__title {
  font-size: $font-size-lg;
  font-weight: $font-weight-bold;
  text-transform: uppercase;
  color: $font-color-white;
}

.button__lang {
  padding: 0;
  height: 20px;
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 10px;
  border: none;
  background-color: transparent;
  cursor: pointer;
  outline: none;
  z-index: 80;
}
