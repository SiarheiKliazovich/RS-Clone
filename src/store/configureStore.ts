/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
