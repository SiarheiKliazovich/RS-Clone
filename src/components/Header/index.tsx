import "./header.scss";
import { FunctionComponent } from "react";
import { Link } from "react-router-dom";
import Profile from "../Profile";
import flagUS from "../../assets/images/jpg/flag-us.jpg";
import flagRU from "../../assets/images/jpg/flag-ru.jpg";

const Header: FunctionComponent = () => {
  // TODO Temporary
  const showProfile = (): void => {
    const profile = document.querySelector(".profile") as HTMLElement;
    profile.classList.toggle("profile--show");
  };

  const language = "en";

  return (
    <header className="header">
      <div className="container">
        <div className="header__wrapper">
          <span className="header__hamburger"></span>
          <Link className="header__link" to="/dashboard">
            <span className="header__logo"></span>
          </Link>
          <ul className="header__list">
            <li className="header__item">
              <div className="language">
                <img
                  className="language__icon"
                  src={language === "en" ? flagUS : flagRU}
                  alt="Current language"
                />
                <span className="language__title">en</span>
              </div>
            </li>
            <li className="header__item" onClick={showProfile}>
              <span className="header__profile"></span>
            </li>
          </ul>
        </div>
      </div>
      <Profile />
    </header>
  );
};
export default Header;
