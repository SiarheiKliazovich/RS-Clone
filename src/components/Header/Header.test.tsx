import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import Header from ".";

const mockStore = configureStore([]);

describe("Header", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      langReducer: {
        lang: "en",
      },
      auth: {
        user: {
          role: "USER",
        },
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <Header
            accentColor={{ static: "#fff", hover: "#000" }}
            isNavbarNightMode={false}
            isButtonVisible={true}
          />
        </Router>
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
