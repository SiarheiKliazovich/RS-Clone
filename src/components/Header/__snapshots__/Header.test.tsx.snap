// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  class="header"
  style="background-color: rgb(255, 255, 255);"
>
  <div
    class="container"
  >
    <div
      class="header__wrapper"
    >
      <span
        class="header__hamburger"
        title="header__hamburger"
      />
      <a
        class="header__link"
        href="/"
      >
        <span
          class="header__logo"
        />
      </a>
      <ul
        class="header__list"
      >
        <li
          class="header__item"
        >
          <div
            class="language"
          >
            <button
              class="button__lang"
              title="button__lang"
            >
              <img
                alt="Current language"
                class="language__icon"
                src="flag-us.jpg"
              />
            </button>
            <span
              class="language__title"
            >
              en
            </span>
          </div>
        </li>
        <li
          class="header__item"
          title="header__profile"
        >
          <span
            class="header__profile"
          />
        </li>
      </ul>
    </div>
  </div>
</header>
`;
